File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    
    dev = []
    idx = 0
    
    pushed = [False for i in range(len(progresses))]
    print(pushed)
    for k in range(10):
        for i, item in enumerate(progresses):
            print(pushed)
            if pushed[i] == False:
                progresses[i] += speeds[i]
                if(progresses[i] >= 100):
                    print("i입니다", i)
                    dev.append(i)
                    pushed[i] = True
                    
        ct = 0
        print("dev!!!!!!!!", dev)
        while True:
            # 만약 지금 배포가능성 있다면
            if idx in dev:
                dev.remove(idx)
                print("pop!!", idx)
                idx += 1
                ct += 1
            else:
                if (ct != 0):
                    answer.append(ct)
                break
            
        if(ct != 0):
            answer.append(ct)
        print("dev", dev)
        print("answer", answer)
                
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> list = new ArrayList<>();
        
        int idx = 0;
        int cnt = 0;
        int len = progresses.length;
        while(idx < len) {

            for(int i = 0; i <len; i++){
                progresses[i] += speeds[i];  
            }
            
            if(progresses[idx] >= 100){
                while(idx < len && progresses[idx] >= 100)                 {
                    idx++;
                    cnt++;
                    
                }
                
                list.add(cnt); 
                cnt = 0;
            }
            
        }
        
        
        int[] ans = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            ans[i] = list.get(i);
        }
        return ans;
    }
}